Cilsil/Cil/Parsers/ExceptionParser.cs

```diff
@@ -0,0 +1,293 @@
+
+            Expression catchVariable = null;
+            if (exceptionType != state.Method.Module.TypeSystem.Object)
+            {
+                if (!state.ExceptionBlockStartToEndOffsets.ContainsKey(endBlockOffset + 3)
+                    && !state.ExceptionBlockStartToEndOffsets.ContainsKey(endBlockOffset + 2))
+                {           
+                    state.PushInstruction(instruction.Next);
+                }
+                state.PushInstruction(instruction, exceptionHandlerNode);
+                instruction = state.PopInstruction();
+
+                /*node2 preds:1 succs:4 exn: Conditional exception handler
+                n$49=_fun___instanceof(n$48:java.io.IOException*,sizeof(t=java.io.IOException):void) [line 55];
+                PRUNE(n$49, true); [line 55];
+                *&amp;CatchVar65:void=n$48 [line 55];*/
+
+                /*node3 preds:1 succs:5 exn: Conditional exception handler -- Connects to outter/finally block code directly
+                n$49=_fun___instanceof(n$48:java.io.IOException*,sizeof(t=java.io.IOException):void) [line 55];
+                PRUNE(!n$49, false); [line 55];*/ 
+
+                // n${i+1}=_fun___instanceof(n${i}:{Exception_Type}*,sizeof(t={Exception_Type}:void)
+                (var exceptionAllocationCall, var objectVariable) = 
+                        CreateExceptionAllocationCall(state, Typ.FromTypeReference(exceptionType));   
+
+                var trueNodeCondition = objectVariable;
+                var falseNodeCondition = new UnopExpression(
+                    UnopExpression.UnopKind.LNot, trueNodeCondition, null);
+                (var exceptionTrueNode, var exceptionFalseNode) =
+                    CreateExceptionBranchConditionNodes(state,
+                                                        trueNodeCondition,
+                                                        falseNodeCondition);
+
+                exceptionTrueNode.Instructions.Insert(0, exceptionAllocationCall);
+                exceptionFalseNode.Instructions.Insert(0, exceptionAllocationCall);
+                
+                // Create catch exception variable expression
+                catchVariable = new LvarExpression(
+                                new LocalVariable(Identifier.CatchIdentifier,
+                                                    state.Method));
+                var catchVarIdentifier = state.GetIdentifier(Identifier.IdentKind.Normal);
+                var catchVarStore = new Store(catchVariable,
+                                              new VarExpression(returnVariable),
+                                              expressionType,
+                                              state.CurrentLocation);
+                exceptionTrueNode.Instructions.Add(catchVarStore);
+
+                RegisterNode(state, exceptionTrueNode);
+                RegisterNode(state, exceptionFalseNode);
+
+                Instruction targetFalse = null;
```

Comment:
call it something like nextInstructionIfNoException

Reply:
Done